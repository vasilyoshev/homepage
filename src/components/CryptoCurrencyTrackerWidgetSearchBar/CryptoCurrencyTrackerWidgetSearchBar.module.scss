.coinSearch {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin-bottom: 8px;
}

.coinText {
  margin: 8px 0;
}

.coinInput {
  width: 260px;
  height: 35px;
  padding-left: 16px;
  border: 1px solid #3d3939;
  border-radius: 5px;
}
