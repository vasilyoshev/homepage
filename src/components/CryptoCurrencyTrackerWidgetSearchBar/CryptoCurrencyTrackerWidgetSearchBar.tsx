import React from 'react';
import { CryptoCurrencyTrackerWidgetSearchBarProps } from 'interfaces';
import styles from './CryptoCurrencyTrackerWidgetSearchBar.module.scss';

export const CryptoCurrencyTrackerWidgetSearchBar: React.FC<CryptoCurrencyTrackerWidgetSearchBarProps> = ({ onChange }) => {
  return (
    <div className={styles.coinSearch}>
      <h1 className={styles.coinText}>Crypto currency</h1>
      <form>
        <input
          className={styles.coinInput}
          type="text"
          placeholder='search...'
          onChange={onChange} />
      </form>
    </div>
  );
};
