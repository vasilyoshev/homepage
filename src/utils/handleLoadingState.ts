import axios from 'axios';
import { decrementLoadingCount, incrementLoadingCount } from 'slices';
import store from 'store';

export const handleLoadingState = (): void => {
  axios.interceptors.response.use(
    (response) => {
      setTimeout(() => store.dispatch(decrementLoadingCount()), 200);

      return response;
    },
    () => {
      store.dispatch(decrementLoadingCount());
    },
  );

  axios.interceptors.request.use((request) => {
    store.dispatch(incrementLoadingCount());

    return request;
  });
};
