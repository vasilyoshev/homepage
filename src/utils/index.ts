export * from './handleLoadingState';
export * from './testUtils';
