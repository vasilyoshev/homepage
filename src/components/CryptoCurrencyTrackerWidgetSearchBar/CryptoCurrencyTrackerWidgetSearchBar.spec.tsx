import { render } from 'utils';
import { CryptoCurrencyTrackerWidgetSearchBar } from './CryptoCurrencyTrackerWidgetSearchBar';

describe('CryptoCurrencyTrackerWidgetSearchBar', () => {
  it('renders successfully', () => {
    const { asFragment } = render(<CryptoCurrencyTrackerWidgetSearchBar type='text' placeholder='Search' onChange={jest.fn()} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
