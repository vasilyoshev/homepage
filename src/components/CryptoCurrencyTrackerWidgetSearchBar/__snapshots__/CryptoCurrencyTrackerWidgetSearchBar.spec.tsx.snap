// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptoCurrencyTrackerWidgetSearchBar renders successfully 1`] = `
<DocumentFragment>
  <div
    class="coinSearch"
  >
    <h1
      class="coinText"
    >
      Crypto currency
    </h1>
    <form>
      <input
        class="coinInput"
        placeholder="search..."
        type="text"
      />
    </form>
  </div>
</DocumentFragment>
`;
